feature_tests/js/api.mjs

```diff
@@ -105,7 +105,187 @@ export class Opaque {
     const diplomat_MyStruct_extracted_d = s["d"];
     const diplomat_MyStruct_extracted_e = s["e"];
     const diplomat_MyStruct_extracted_f = s["f"];
-    const diplomat_out = wasm.Opaque_assert_struct(this.underlying, diplomat_MyStruct_extracted_a, diplomat_MyStruct_extracted_b, diplomat_MyStruct_extracted_c, diplomat_MyStruct_extracted_d, diplomat_MyStruct_extracted_e, diplomat_MyStruct_extracted_f);
+    if (!diplomat_MyStruct_extracted_f.length || !diplomat_MyStruct_extracted_f.codePointAt || diplomat_MyStruct_extracted_f.length != 1) { throw new TypeError("Expected single-character string for `char` parameter diplomat_MyStruct_extracted_f, found " + diplomat_MyStruct_extracted_f); }
+    const diplomat_out = wasm.Opaque_assert_struct(this.underlying, diplomat_MyStruct_extracted_a, diplomat_MyStruct_extracted_b, diplomat_MyStruct_extracted_c, diplomat_MyStruct_extracted_d, diplomat_MyStruct_extracted_e, diplomat_MyStruct_extracted_f.codePointAt(0));
+  }
+}
+
+const OptionOpaque_box_destroy_registry = new FinalizationRegistry(underlying => {
+  wasm.OptionOpaque_destroy(underlying);
+});
+
+export class OptionOpaque {
+  constructor(underlying) {
+    this.underlying = underlying;
+  }
+
+  static new(i) {
+    const diplomat_out = (() => {
+      if (wasm.OptionOpaque_new(i) !== 0) {
+        const inhabited_value = (() => {
+          const out = (() => {
+            const out = new OptionOpaque(wasm.OptionOpaque_new(i));
```

Comment:
Question: why are you calling `wasm.OptionOpaque_new(i)` twice?  Shouldn't you call it once, on line 124, and save the value for down here?